File: app/android/app/src/main/java/com/jun/weather/viewmodel/entity/WeatherPointModel.java
package com.jun.weather.viewmodel.entity;

import androidx.annotation.NonNull;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class WeatherPointModel {
    public HashMap<String, Integer[]> indexMap = new HashMap<>();
    public List<WeatherPoint> weatherPointList = new ArrayList<>();

    public static class WeatherPoint {
        public String deg_1="",deg_2="",deg_3="";
        public int x,y;
        //temp
        public String midTempCode;
        //weather
        public String midWeatherCode;

        @Override
        @NonNull
        public String toString() {
            return deg_1+(deg_2.isEmpty()?"":" "+deg_2)+(deg_3.isEmpty()?deg_3:" "+deg_3);
        }
    }
}

File: app/android/app/src/main/java/com/jun/weather/ui/entity/WeatherPointModel.kt
package com.jun.weather.ui.entity

import java.lang.StringBuilder
import java.util.*

data class WeatherPointModel(
        var indexMap: HashMap<String, Array<Int>> = HashMap<String, Array<Int>>(),
        var weatherPointList: MutableList<WeatherPoint> = mutableListOf(),
)

data class WeatherPoint(
    var deg_1: String = "",
    var deg_2: String = "",
    var deg_3: String = "",
    var x: Int = 0,
    var y: Int = 0,
    //temp
    var midTempCode: String? = null,

    //weather
    var midWeatherCode: String? = null,
) {
    override fun toString(): String {
        val sb = StringBuilder(deg_1)
        if(deg_2.isNotEmpty()) {
            sb.append(" $deg_2")
        }
        if(deg_3.isNotEmpty()) {
            sb.append(" $deg_3")
        }
        return sb.toString()
    }
}